pennylane/ops/op_math/exp.py

```diff
@@ -224,7 +225,7 @@ def has_decomposition(self):
             coeff *= base.scalar
             base = base.base
         is_pauli_rot = qml.pauli.is_pauli_word(self.base) and math.real(self.coeff) == 0
-        is_hamiltonian = isinstance(base, Hamiltonian)
+        is_hamiltonian = isinstance(base, (Hamiltonian, LinearCombination))
```

Comment:
Definitely a location where an `Operator.num_terms` property would be really useful.